examples/mixed_hmm/model.py

```diff
@@ -0,0 +1,527 @@
+        with interpretation(eager):
+            eps_i_dist = plate_g + plate_i + dist.Normal(**params["eps_i"])(value=eps_i)
+
+        log_prob.append(eps_i_dist)
+    else:
+        eps_i = to_funsor(0.)
+
+    # add group-level and individual-level random effects to gamma
+    with interpretation(eager):
+        gamma = gamma + eps_g + eps_i
+
+    # initialize y in a single state for now
+    y = Number(0, config["sizes"]["state"])
+
+    N_t = config["sizes"]["timesteps"]
+    N_state = config["sizes"]["state"]
+    for t in range(N_t):
+
+        gamma_t = gamma  # per-timestep variable
+
+        # we've accounted for all effects, now actually compute gamma_y
+        gamma_y = gamma_t(y_prev=y)
+
+        y = Variable("y_{}".format(t), bint(N_state))
+        with interpretation(eager):
+            y_dist = plate_g + plate_i + dist.Categorical(
+                probs=gamma_y.exp() / gamma_y.exp().sum()
+            )(value=y)
+
+        log_prob.append(y_dist)
+
+        # observation 1: step size
+        with interpretation(eager):
+            step_zi = dist.Categorical(probs=params["zi_step"]["zi_param"](y_curr=y))(
+                value=Tensor(zi_masks["step"].data, zi_masks["step"].inputs, 2)(t=t))
+            step_dist = plate_g + plate_i + step_zi + dist.Gamma(
+                **{k: v(y_curr=y) for k, v in params["step"].items()}
+            )(value=observations["step"](t=t)) * zi_masks["step"](t=t)
+
+        log_prob.append(step_dist)
+
+        # observation 2: step angle
+        with interpretation(eager):
+            angle_dist = plate_g + plate_i + dist.VonMises(
+                **{k: v(y_curr=y) for k, v in params["angle"].items()}
+            )(value=observations["angle"](t=t))
+
+        log_prob.append(angle_dist)
+
+        # observation 3: dive activity
+        with interpretation(eager):
+            omega_zi = dist.Categorical(probs=params["zi_omega"]["zi_param"](y_curr=y))(
+                value=Tensor(zi_masks["omega"].data, zi_masks["omega"].inputs, 2)(t=t))
+            omega_dist = plate_g + plate_i + omega_zi + dist.Beta(
+                **{k: v(y_curr=y) for k, v in params["omega"].items()}
+            )(value=observations["omega"](t=t)) * zi_masks["omega"](t=t)
+
+        log_prob.append(omega_dist)
+
+    return log_prob
+
+
+def model_parallel(config):
+    """
+    Simpler version of generic model with no zero-inflation
+    """
+
+    N_state = config["sizes"]["state"]
+
+    params = initialize_model_params(config)
+    observations = initialize_observations(config)
+    raggedness_masks = initialize_raggedness_masks(config)
+    zi_masks = initialize_zi_masks(config)
+
+    # initialize gamma to uniform
+    gamma = Tensor(
+        torch.zeros((N_state, N_state)),
+        OrderedDict([("y_prev", bint(N_state))]),
+    )
+
+    N_v = config["sizes"]["random"]
+    N_c = config["sizes"]["group"]
+    log_prob = []
+
+    plate_g = Tensor(torch.zeros(N_c), OrderedDict([("g", bint(N_c))]))
```

Comment:
note: This is an interesting pattern, denoting plates as zero `Tensor`s. I think we could avoid this after #233 , since plate size can be determined from context. (unrelated to this PR, but it would be nice cleanup in the future).